### ISSUE TITLE
User can be playing in multiple rooms

### ISSUE THREAD

**Issue by lchen025** [xnor-gate/shibboleth_bot#4]
This probably shouldn't be allowed? It also makes the player list display weird, because they will show up as playing in an arbitrary room (whichever is first in the roles list).

Should add a check upon joining that the player isn't in any other room's player list, or if they are in another queue they should be removed from it.

**Comment by xnor-gate** [comment]
I agree this is not ideal, and was wondering about a better way to do this. Does every room needs to be checked whenever a player joins? Or maybe just within the same server (guild)?  Currently each room accesses only things within it and its channel. Calling the unjoin command for them for each other room they are in might be nice to let the existing functionality handle queues and posting a message.

**Comment by lchen025** [comment]
I'll look more tomorrow, but my first thought is to add a class that manages state for a user, and that can enforce stuff like they can't be playing in more than one game, can't be in multiple join queues etc
